python/convert_to_dataset.py

```diff
@@ -0,0 +1,128 @@
+"""Convert Synapse Folder to Dataset
+
+TODO:
+  - use Python client once Dataset support becomes available
+  - get Folder annotations and add them to Dataset entity
+  - apiKey will be deprecated, update to PAT in login()
+
+author: verena.chung
+"""
+
+import os
+import json
+import getpass
+
+import synapseclient
+from synapseclient import Column
+
+PARENT_ID = "syn21498902"
+FOLDERS_ID = "syn22047105"
+
+
+def login():
+    """Log into Synapse. If env variables not found, prompt user.
+
+    Returns:
+        syn: Synapse object
+    """
+    try:
+        syn = synapseclient.login(
+            os.getenv('SYN_USERNAME'),
+            apiKey=os.getenv('SYN_APIKEY'),
+            silent=True)
+    except synapseclient.core.exceptions.SynapseNoCredentialsError:
+        print("Credentials not found; please manually provide your",
+              "Synapse username and password.")
+        username = input("Synapse username: ")
+        password = getpass.getpass("Synapse password: ")
+        syn = synapseclient.login(username, password, silent=True)
+    return syn
+
+
+def create_dataset_schema(syn):
+    """Create schema for creating a Dataset entity.
+
+    Returns:
+        list of ColumnModel IDS
+    """
+    cols = syn.createColumns(
+        columns=[
+            Column(name="id", columnType="ENTITYID"),
+            Column(name="fileName", columnType="STRING", maximumSize=256),
+            Column(name="name", columnType="STRING", maximumSize=256),
+            Column(name="species", columnType="STRING", maximumSize=25),
+            Column(name="dataFormat", columnType="STRING", maximumSize=15),
+            Column(name="assay", columnType="STRING", maximumSize=500),
+            Column(name="tumorType", columnType="STRING", maximumSize=500),
+            Column(name="gender", columnType="STRING", maximumSize=11)
+        ]
+    )
+    return [col.get('id') for col in cols]
+
+
+def convert_data_files(syn, folder_id):
+    """Convert `File` entities into `DatasetItem` entities.
+
+    Assumptions:
+        Immediate children of given Folder ID are File entities.
+
+    Returns:
+        list of `DatasetItem`s
+    """
+    files = syn.getChildren(folder_id)
+    return [
+        {'entityId': f.get('id'), 'versionNumber': f.get('versionLabel')}
+        for f in files
+    ]
+
+
+def convert_folder_to_dataset(syn, dataset_name, schema, dataset_items):
```

Comment:
The name of this function is semi misleading.  I thought you were literally taking a Folder entity and converting the Folder Entity to a Dataset.  But you're creating a Dataset with the same name as the folder name.  `_create_dataset()` is more clear.

Reply:
Sounds good, thanks!